File: app/src/main/java/com/lostincontext/ruledetails/RuleDetailsPresenterModule.java
package com.lostincontext.ruledetails;

import com.lostincontext.awareness.AwarenessModule;

import dagger.Module;
import dagger.Provides;


@Module(includes = {AwarenessModule.class})
public class RuleDetailsPresenterModule {

    private final RuleDetailsContract.View view;

    public RuleDetailsPresenterModule(RuleDetailsContract.View view) {
        this.view = view;
    }

    @Provides
    RuleDetailsContract.View provideMainScreenContractView() {
        return view;
    }
}

File: app/src/main/java/com/lostincontext/ruledetails/RuleDetailsPresenterModule.kt
package com.lostincontext.ruledetails

import android.os.Bundle
import com.lostincontext.awareness.AwarenessModule

import dagger.Module
import dagger.Provides


@Module(includes = arrayOf(AwarenessModule::class))
class RuleDetailsPresenterModule(private val view: RuleDetailsContract.View,
                                 private val icicle: Bundle?) {

    @Provides internal fun provideView(): RuleDetailsContract.View = view

    @Provides internal fun provideBundle(): Bundle? = icicle

}
